layouts/partials/nav.html

```diff
@@ -64,10 +64,55 @@
             {{ end }}
 
             <li class="ms-2">
-                <button class="btn donate-btn btn-outline-primary text-uppercase px-4 rounded-0"
-                        type="button" style="font-size: 0.9rem;">
+                <button class="btn donate-btn btn-outline-primary text-uppercase px-4 rounded-0" type="button" data-bs-toggle="modal" data-bs-target="#donateModal" style="font-size: 0.9rem;">
                     Donate
-                </button>
+                  </button>
+                  
+                  <!-- Donate Modal -->
+                <div class="modal fade" id="donateModal" tabindex="-1" aria-labelledby="donateModalLabel" aria-hidden="true">
+                <div class="modal-dialog">
+                    <div class="modal-content">
+                    <div class="modal-header">
+                        <h5 class="modal-title" id="donateModalLabel">Donate Now!</h5>
+                        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+                    </div>
+                    <div class="modal-body">
+                        <!-- donation content here --> 
+                    <p style="font-size: 18px;">Thank you for considering a donation. <br>
+                        Please transfer to:</p> 
+                    <p style="line-height: 1.2; margin-bottom: 0; font-size: 21px;" >
+                        3233325318<br>
+                        PPBP MALAYSIA<br>
+                        PUBLIC BANK
+                    </p>
+                    <img src="{{ absURL "TIAPaymentQR.jpeg" }}" alt="QR Code" style="width: 100%; height: auto; margin-top: 20px;">
+                    </div>
+                    
+                    <div class="modal-footer">
+                        <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
+                        
+                    </div>
```

Comment:
```suggestion
```

I suggest no footer, due to:
- the img takes up quite a lot of space, it would be quite crowded
- we have a close button alr at the top, it is redundant

Reply:
Another thing that u may consider:

Prevent the modal from closing if clicking outside.

![image](https://github.com/user-attachments/assets/e4338486-87ba-4f93-a439-930106e19d10)

I'm not sure whether this is a good decision...? I will leave it to you.

Reply:
The reason I left this out initially is because I thought it would be better if users could click anywhere to exit out of convenience. I will discuss with the team in the future and add this if necessary, hope that is okay